lib/install/opsuser/opsuser.go

```diff
@@ -184,11 +199,21 @@ func (mgr *RBACManager) setupPermissions(ctx context.Context) ([]rbac.ResourcePe
 		resourceDescs = append(resourceDescs, resourceDesc{rbac.Network, *netRef})
 	}
 
-	// Add resource pools
+	// Add the endpoint role to the resource pool(s).
 	for _, rPoolRef := range mgr.configSpec.ComputeResources {
 		resourceDescs = append(resourceDescs, resourceDesc{rbac.Endpoint, rPoolRef})
 	}
 
+	// For vCenter, apply the endpoint role to the VCH inventory folder as well.
+	if mgr.session.IsVC() {
```

Comment:
Should we ensure the `VCHFolder` isn't the `VmFolder` and only apply if not the `VmFolder`?

Reply:
Thanks - @matthewavery brought this up too.

On VC, we're setting `VCHFolder` [here](https://github.com/vmware/vic/pull/7587/files#diff-13cb124a8b8b2d0e7744466165fd84ffR522) only when folder creation is successful. If it isn't, we'll error out before getting to the ops-user code.

Also, this code is only invoked in the VCH create path so going forward we can assert that the VCH will be in its own folder.

Reply:
Yeah, another part of our conversation @cgtexmex is what we should do if that was the case. We do not want to give the VMFolder the priveledges, but everything is already deployed so we would likely have to warn that the VCH is broken and requires a delete. or we could attempt the delete on our own. But this is past the point where everything else has been set up(tmk).